server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -278,6 +279,28 @@ public void getSynonymRule(String synonymSetId, String synonymRuleId, ActionList
         );
     }
 
+    public void deleteSynonymRule(
+        String synonymSetId,
+        String synonymRuleId,
+        ActionListener<SynonymsReloadResult<AcknowledgedResponse>> listener
+    ) {
+        checkSynonymSetExists(
+            synonymSetId,
+            listener.delegateFailure(
+                (l1, obj) -> client.prepareDelete(SYNONYMS_ALIAS_NAME, internalSynonymRuleId(synonymSetId, synonymRuleId))
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                    .execute(new DelegatingIndexNotFoundActionListener<>(synonymSetId, l1, (l2, deleteResponse) -> {
+                        if (deleteResponse.getResult() == DocWriteResponse.Result.NOT_FOUND) {
+                            l2.onFailure(new ResourceNotFoundException("synonym rule [" + synonymRuleId + "] not found"));
+                            return;
+                        }
+
+                        reloadAnalyzers(synonymSetId, l2, AcknowledgedResponse.of(true));
```

Comment:
Is there any use case for putting `reloadAnalyzers` behind a parameter? For example, if I have 100 synonyms to remove I may want to reload analyzers only once?

Reply:
@kderusso Carlos and me explored an idea of having CRUD operations with a parameter `reload_analyzers=false`.  The idea is that a user issues a number of CRUD requests, and reload analyzers after all of them only once. But we have not implemented this idea.